### DISCUSSION TITLE
2 GPU is 10 times slower than 1 GPU, using colossal, normal?

### DISCUSSION THREAD

**Discussion by Arsmart123** [hpcaitech/ColossalAI#290] [Community | Q&A]
Hi! I am comparing the difference between split a single layer into two GPUs, and my result is like:
2GPUs single layer with parameter: 1,600,000*4000(about...) => 111s
1 GPU single layer with same parameter => 9s
Such delay is normal? Thank you!

**Comment by FrankLeeeee** [comment]
Hi, do you have a code snippet to reproduce this issue?

**Reply by Arsmart123** [reply]
Thank you very much!!! I will try that tomorrow!

**Reply by Arsmart123** [reply]
https://github.com/Arsmart123/colossal-time-question
I post a link here, if you could check it and help me, it will be highly appreciated!!!!

**Reply by FrankLeeeee** [reply]
Great, let me try to reproduce this problem.

**Comment by FrankLeeeee** [comment]
Hi @Arsmart123 , I have run your code. The time taken is 2.67s for one process and 2.55s for two processes.

**Reply by FrankLeeeee** [reply]
I can't reproduce your performance. Can you tell me more details about your environment?

**Reply by FrankLeeeee** [reply]
I also notice that this model is rather small. When a model can fit in one GPU, there is no need to introduce tensor parallel.

**Reply by Arsmart123** [reply]
> I can't reproduce your performance. Can you tell me more details about your environment?
Hi! I really can not install the latest colossalai in NUS HPC, so I am using a older verison, below is the link:
https://github.com/Arsmart123/colossal-time-question/blob/main/ColossalAI-main-%E6%97%A7%E7%89%88%E5%A4%87%E4%BB%BD.zip

Thank you very much!!!!

**Reply by Arsmart123** [reply]
> I can't reproduce your performance. Can you tell me more details about your environment?
Also, I am using torch 1.10 cu11.3
